pilot/pkg/networking/plugin/plugin.go

```diff
@@ -59,14 +56,22 @@ type Plugin interface {
 	// Can be used to add additional filters.
 	OnInboundListener(in *InputParams, mutable *istionetworking.MutableObjects) error
 
-	// OnInboundFilterChains is called whenever a plugin needs to setup the filter chains, including relevant filter chain
-	// configuration, like FilterChainMatch and TLSContext.
-	OnInboundFilterChains(in *InputParams) []istionetworking.FilterChain
-
 	// OnInboundPassthrough is called whenever a new passthrough filter chain is added to the LDS output.
 	// Can be used to add additional filters.
 	OnInboundPassthrough(in *InputParams, mutable *istionetworking.MutableObjects) error
 
-	// OnInboundPassthroughFilterChains is called whenever a plugin needs to setup custom pass through filter chain.
-	OnInboundPassthroughFilterChains(in *InputParams) []istionetworking.FilterChain
+	// InboundMTLSConfiguration configures the mTLS configuration for inbound listeners.
+	InboundMTLSConfiguration(in *InputParams, passthrough bool) []MTLSSettings
+}
+
+// MTLSSettings describes the mTLS options for a filter chain
+type MTLSSettings struct {
+	// Port is the port this option applies for
+	Port uint32
+	// Mode is the mTLS  mode to use
+	Mode model.MutualTLSMode
+	// TCPTLSContext describes the tls context to use for TCP filter chains
+	TCPTLSContext *tls.DownstreamTlsContext
```

Comment:
Just calling them TCP/HTTP is fine since TLSContext is already derived from struct name?